Cost_Functions/cost_function_wrapper.py

```diff
@@ -1,32 +1,41 @@
 from importlib import import_module
 import os
+
+from Control_Toolkit.others.globals_and_utils import get_logger
 from SI_Toolkit.computation_library import TensorType
 import yaml
 from copy import deepcopy as dcp
 from types import MappingProxyType
 from Control_Toolkit.Controllers import template_controller
 
 from Control_Toolkit.Cost_Functions import cost_function_base
+from others.globals_and_utils import load_or_reload_config_if_modified, update_attributes
 
-# cost_function config
-cost_function_config = yaml.load(
-    open(os.path.join("Control_Toolkit_ASF", "config_cost_function.yml"), "r"),
-    Loader=yaml.FullLoader,
-)
-
+log=get_logger(__name__)
 
 class CostFunctionWrapper:
     def __init__(self):
         self.cost_function = None
-        self.cost_function_name_default: str = cost_function_config[
-            "cost_function_name_default"
-        ]
+        # cost_function config
+        (self.cost_function_config, _) = load_or_reload_config_if_modified(
+            os.path.join("Control_Toolkit_ASF", "config_cost_functions.yml"))
+
+        self.cost_function_name_default: str = self.cost_function_config.cost_function_name_default
+        self.lib = None # filled by configure(), needed to update TF variables
+
+        log.info(f'default cost function name is {self.cost_function_name_default}')
 
     def configure(
         self,
         controller: template_controller,
         cost_function_specification=None,
     ):
+        """
+        Configures the cost function. TODO This lazy constructor is needed why?
```

Comment:
I agree actually. At this point I don't see a reason for the wrapper any(more).

The only justification was: You can create a cost function wrapper without knowing at that time which cost function (default, trajectory follower, boundary grad, etc.) to use.

However, the only location where this object is instantiated, is in the controller constructor. There, the cost function information is available.